cmd/scollector/collectors/local_listener.go

```diff
@@ -0,0 +1,106 @@
+package collectors
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"bosun.org/_third_party/github.com/MiniProfiler/go/miniprofiler"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+	"bosun.org/util"
+)
+
+func LocalListener(listenAddr string) error {
+	collectors = append(collectors, &StreamCollector{F: func() <-chan *opentsdb.MultiDataPoint {
+		return c_local_listener(listenAddr)
+	},
+		name: fmt.Sprintf("local_listener-%s", listenAddr),
+	})
+
+	return nil
+}
+
+func c_local_listener(listenAddr string) <-chan *opentsdb.MultiDataPoint {
+	pm := &putMetric{}
+	pm.localMetrics = make(chan *opentsdb.MultiDataPoint, 1)
+
+	mux := http.NewServeMux()
+	mux.Handle("/api/put", pm)
+	mux.Handle("/api/metadata/put", util.JSON(PutMetadata))
```

Comment:
I don't think you really need the json middleware for this. You already do error handling, and there is no response to serialize. Probably should revert moving of that func.